### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by lcreid** [bootstrap-ruby/bootstrap_form#492]
It seems to us like `bootstrap_form` could provide support to make Rails forms more accessible to people with different levels of ability in sight, mobility, and other aspects of computer interaction. This is likely a large project that would be undertaken via a number of smaller pull requests. 

We invite you to create individual issues for parts of this effort. Please reference this issue in your issue, so that we can keep track of all the work being done. Once an implementation approach has been confirmed, make a pull request via the [usual process](/bootstrap-ruby/bootstrap_form/blob/master/CONTRIBUTING.md).

Questions and comments on this idea should be discussed here in this issue.

To support this idea, we've collected some information from the Internet:

Some links:

- https://www.w3.org/TR/wai-aria-practices-1.1/. Note that one of the first sections of this document is titled, "No ARIA is better than bad ARIA".
- https://www.w3.org/TR/wai-aria-practices-1.2/. The working draft of the next version of the above document.
- https://www.w3.org/TR/WCAG20/
- U.S. Government specs: https://www.usability.gov/what-and-why/accessibility.html.
- Canadian Government specs: https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=23601.
- ESPN a11y guide: http://www.espn.com/core/toolkit/page/webAccessibilityGuide.
- Someone's checklist: https://a11yproject.com/checklist.html.
- An es-lint plugin for accessbility: https://www.npmjs.com/package/eslint-plugin-jsx-a11y.

Some free screen readers:

- [VoiceOver](https://www.apple.com/ca/accessibility/mac/vision/). Mac screen reader.
- [NVDA](https://www.nvaccess.org/download/). Windows screen reader.
- [Orca](http://projects.gnome.org/orca/). The screen reader bundled in Ubuntu. It's a Gnome project, so should be easy to use on any Gnome distro, and probably usable on others as well.
- [WebAnywhere](http://webanywhere.cs.washington.edu/). Browser-based screen reader. Advertised as OS-independent.
- usabilitygeek.com has a [list of free assistive technologies](https://usabilitygeek.com/10-free-screen-reader-blind-visually-impaired-users/).

 (I haven't tried any of the screen readers.)

Some podcasts. These podcasts may contain material that's specific to technologies other than Rails, but they still contain a lot of relevant information:

- https://modernweb.podbean.com/e/s04e12-web-a11y-with-jen-luker-and-necoline-hubner/.
- http://www.fullstackradio.com/97.
- https://www.digitala11y.com/web-accessibility-podcasts-rountup/. This is a list of other podcasts.
